packages/blockfrost/src/BlockfrostToCore.ts

```diff
@@ -26,16 +26,16 @@ export const BlockfrostToCore = {
   currentWalletProtocolParameters: (
     blockfrost: Responses['epoch_param_content']
   ): ProtocolParametersRequiredByWallet => ({
-    coinsPerUtxoByte: Number(blockfrost.coins_per_utxo_word),
-    maxCollateralInputs: Number(blockfrost.max_collateral_inputs),
-    maxTxSize: Number(blockfrost.max_tx_size),
-    maxValueSize: Number(blockfrost.max_val_size),
-    minFeeCoefficient: blockfrost.min_fee_a,
-    minFeeConstant: blockfrost.min_fee_b,
-    minPoolCost: Number(blockfrost.min_pool_cost),
-    poolDeposit: Number(blockfrost.pool_deposit),
+    coinsPerUtxoByte: BigInt(blockfrost.coins_per_utxo_word ?? 0),
```

Comment:
This can lead to weird bugs. It would be better to throw an error (BigInt already does that, and blockfrost impl already wraps it in ProviderError)

```suggestion
    coinsPerUtxoByte: BigInt(blockfrost.coins_per_utxo_word!),
```

Reply:
@mkazlauskas you are correct. The thing is that we shouldn't throw an error here (according to the interface); these fields are optional, previously the Number constructor was being used and it was converting undefined values to 'NaN'; however; bigint has no NaN equivalent so I defaulted it to 0 (which is incorrect as you pointed out), this si what I did to correct this:

https://github.com/input-output-hk/cardano-js-sdk/pull/388/commits/c6d7dc4b26a5a050931020e910b6af6ea9730397

Reply:
I still think it should throw since our `ProtocolParametersRequiredByWallet` has every field as required. The fix you made in c6d7dc4b26a5a050931020e910b6af6ea9730397 can return an object that does not conform to the return type - we should try avoid that if we can.

It should only be undefined in earlier eras. For a wallet, requiring them to be always defined should be fine.

Reply:
Ok I see it now, some of the test are omitting some of the fields on the mocks and I saw before we had this:

```
export type ProtocolParametersBabbage = OptionalUndefined<
  RecursivelyReplaceNullWithUndefined<Ogmios.ProtocolParametersBabbage>
>;
```

So I translated that to this:

```
export interface ProtocolParametersBabbage {
  minFeeCoefficient?: bigint;
  minFeeConstant?: bigint;
  maxBlockBodySize?: bigint;
  maxBlockHeaderSize?: bigint;
  maxTxSize?: bigint;
  stakeKeyDeposit?: bigint;
  poolDeposit?: bigint;
  poolRetirementEpochBound?: bigint;
  desiredNumberOfPools?: bigint;
  poolInfluence?: number;
  monetaryExpansion?: number;
  treasuryExpansion?: number;
  minPoolCost?: bigint;
  coinsPerUtxoByte?: bigint;
  maxValueSize?: bigint;
  collateralPercentage?: bigint;
  maxCollateralInputs?: bigint;
  protocolVersion?: ProtocolVersion;
  costModels?: CostModels;
  prices?: Prices;
  maxExecutionUnitsPerTransaction?: ExUnits;
  maxExecutionUnitsPerBlock?: ExUnits;
}
```

I alos missed the `Required<T>`, I will make the fields required them and fix the tests

Reply:
Right, omitting unused fields in tests is fine.

> export type ProtocolParametersBabbage = OptionalUndefined<

This should probably be updated to have required fields, but let's just leave it as-is for now.

Reply:
Okay I think now it should be appropriately fixed: https://github.com/input-output-hk/cardano-js-sdk/pull/388/commits/a5b17a79f1044221c5a3b13d7c76fe74210089c1

The mock cant ommit any value now since we are making all fields required (it was a mock blockfrost response).